### PR TITLE
Fix: don't open stdin if versionlock is missing

### PR DESCRIPTION
**Pull request by inknos**
BZ: https://bugzilla.redhat.com/show_bug.cgi?id=1785563
Tests: https://github.com/rpm-software-management/ci-dnf-stack/pull/783

### CODE REVIEW

**Comment by Conan-Kudo** [comment]
@inknos Can you please fix the authorship of your first commit? It looks like it has your computer's username and hostname instead of your correct identity.

**Comment by pkratoch** [comment]
There are actually two fixes, right? One is that the messages are printed on stdout instead of logged and that fixes the bug 1785563.

The other one is to error out if ``locklist_fn`` is not set. The problem with this is that when I have versionlock installed and I delete the ``locklist`` option from ``versionlock.conf``, then I get an error even when I run ``dnf upgrade``. Which I don't think is desirable. We should distinguish between running the versionlock command (in that case, it should error out) and running some other command (in that case, there should be just a warning, such as "versionlock enabled, but locklist not set"). 

Also, we don't check if ``locklist_fn`` is set in the ``_write_locklist`` (running ``dnf versionlock add libdnf`` prints that it adds libdnf, but it actually doesn't have any effect), so if you could add it, too, that would be great.

File: plugins/versionlock.py [hunk 1]
```diff
@@ -192,6 +192,8 @@ def run(self):
 def _read_locklist():
     locklist = []
     try:
+        if not locklist_fn:
+            raise dnf.exceptions.Error(NO_LOCKLIST)
```
Comment (inknos) [inline plugins/versionlock.py]: These lines just check if the filename is valid. If you are not checking the file name you can try to open a `NaN` which will reault in opening a input stream.  
> The problem with this is that when I have versionlock installed and I delete the locklist option from versionlock.conf, then I get an error even when I run dnf upgrade. Which I don't think is desirable

This is actually expected

Comment (inknos) [reply plugins/versionlock.py]: >Also, we don't check if locklist_fn is set in the _write_locklist (running dnf versionlock add libdnf prints that it adds libdnf, but it actually doesn't have any effect), so if you could add it, too, that would be great.

I will add the check inside the `_write_locklist` as it is in `_read_locklist`

**Comment by pkratoch** [comment]
There was another change to versionlock plugin recently: https://github.com/rpm-software-management/dnf-plugins-core/pull/376
Surprisingly, there are no conflicts, but it changes the behaviour because for "add" and "exclude" it reads the locklist before writing into it, so with missing locklist, it raises an error instead of creating new file.
Can you please rebase to newest master and fix this problem?

**Comment by pkratoch** [comment]
@rh-atomic-bot r+

**Pull request closed without merging**